lib/at_parser/at_match.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file at_match.h
+ *
+ * @defgroup at_match Internal AT parser matching functions.
+ * @ingroup at_parser
```

Comment:
Is the thought to include the functions in this header to documentation or to leave them out?

Reply:
Ah, good point.

Reply:
I think it's fine to leave it as is, `at_utils.h` in the `at_cmd_parser` also has a similar structure and it's supposed to be an internal header only.

Reply:
Yeah. Should be fine. I learned that `:inner:` need to be added after the `.. doxygengroup:: name_of_group` in the `.rst` to show content from subgroups that uses `@ ingroup`.